In [2]:
import arcpy

## Problem 1

In [12]:
arcpy.env.workspace = r"C:\Users\brend_000\Documents\ArcGIS\Projects\GIS 5090\GIS 5090 - Project 1\hurricane_nate_subset"
dems = arcpy.ListRasters()
print(len(dems))

16


Set the workspace to the folder with the tiff files in it

Create a list of all the rasters within said folder

Prints how many rasters are in the folder using the len function to return the number

1. There are 16 rasters in the folder

In [14]:
for dem in dems:
    desc = arcpy.Describe(dem)
    if desc.spatialreference.name == "GCS_WGS_1984":
        print(desc.name + " uses projection " + desc.spatialreference.name)
    else:
        print(desc.name + " does not use 1984, it uses projection " + desc.spatialreference.name)

20171010aC0874030w301630n.tif uses projection GCS_WGS_1984
20171010aC0874200w301500n.tif uses projection GCS_WGS_1984
20171010aC0874200w301630n.tif uses projection GCS_WGS_1984
20171010aC0874330w301500n.tif uses projection GCS_WGS_1984
20171010aC0874330w301630n.tif uses projection GCS_WGS_1984
20171010aC0874500w301500n.tif uses projection GCS_WGS_1984
20171010aC0874630w301500n.tif uses projection GCS_WGS_1984
20171010aC0874800w301500n.tif uses projection GCS_WGS_1984
20171010aC0874930w301500n.tif uses projection GCS_WGS_1984
20171010aC0875100w301500n.tif uses projection GCS_WGS_1984
20171010aC0875230w301500n.tif uses projection GCS_WGS_1984
20171010aC0875400w301500n.tif uses projection GCS_WGS_1984
20171010aC0875530w301500n.tif uses projection GCS_WGS_1984
20171010aC0875700w301500n.tif uses projection GCS_WGS_1984
20171010aC0875830w301500n.tif uses projection GCS_WGS_1984
20171010aC0880000w301330n.tif uses projection GCS_WGS_1984


Create a for loop to search through the rasters in the list

Use the describe function to return describe elements from object properties so we can find the projection name

Use an if statement to see if the rasters all use the same projection

2. GCS_WGS_1984
3. They all use the same projection

In [16]:
for dem in dems:
    desc = arcpy.Describe(dem)
    print(desc.bandCount)

4
4
4
4
4
4
4
4
4
4
4
4
4
4
4
4


For loop searchs through all the raster images in the list
Use the describe function to retrieve info of each raster
bandCount returns the number of bands each raster has
4. 4 bands each

In [17]:
for dem in dems:
    desc = arcpy.Describe(dem)
    print(desc.format)

TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF
TIFF


For loop rotates through the list

Describe function retrieves info of each raster

Format function returns the function of each raster

5. The rasters format is a TIFF

## Problem 2

In [18]:
arcpy.env.workspace = r"C:\Users\brend_000\Documents\ArcGIS\Projects\GIS 5090\GIS 5090 - Project 1\tlgdb_2019_a_us_school.gdb"

In [19]:
dems = arcpy.ListFeatureClasses()

In [20]:
for dem in dems:
    desc = arcpy.Describe(dem)
    print(desc.name)

School_District_Unified
School_District_Secondary
School_District_Elementary
State


Set the workspace to the gdb with all the feature classes

Create the dems object to hold a list of all the feature classes

Create a for loop that loops through the list of each feature class, and the name function returns the feature class name thanks to the describe function being set beforehand

1. School_District_Unified, School_District_Secondary, School_District_Elementary, State

In [25]:
for dem in dems:
    desc = arcpy.Describe(dem)
    print(desc.name + " has " + str(arcpy.GetCount_management(dem)) + " features")

School_District_Unified has 10887 features
School_District_Secondary has 482 features
School_District_Elementary has 1946 features
State has 56 features


The getCount function allows arcpy to return the value of the number of feature classes within a feature class, by returning the total number of rows for a table

2. SDU has 10887 features, SDS has 482, SDE has 1946, and State has 56

In [28]:
for dem in dems:
    desc = arcpy.Describe(dem)
    print(desc.name + " uses projection " + desc.spatialreference.name)

School_District_Unified uses projection GCS_North_American_1983
School_District_Secondary uses projection GCS_North_American_1983
School_District_Elementary uses projection GCS_North_American_1983
State uses projection GCS_North_American_1983


The spatialreference.name function returns the name of the projection used by the feature class currently being looped in the for loop
3. All feature classes use the GCS_North_American_1983 projection

In [29]:
for dem in dems:
    desc = arcpy.Describe(dem)
    print(desc.name + " has shape type " + desc.shapeType)

School_District_Unified has shape type Polygon
School_District_Secondary has shape type Polygon
School_District_Elementary has shape type Polygon
State has shape type Polygon


The shapeType function returns the shape type of the feature class currently being looped in the for loop
4. All of the feature classes are the Polygon shape type

## Problem 3

In [41]:
arcpy.env.workspace = r"C:\Users\brend_000\Documents\ArcGIS\Projects\GIS 5090\GIS 5090 - Project 1\usa_cities.gdb"
fc = "largest_cities"
fields = ["NAME", "POPULATION"]

In [49]:
for row in sorted(arcpy.da.SearchCursor(fc, fields)):
    if row[1] > 1000000:
        print(u'{0} has a population of {1}'.format(row[0], row[1]))
    
    
            

Chicago has a population of 2781116
Dallas has a population of 1323651
Houston has a population of 2333285
Los Angeles has a population of 3986442
New York has a population of 8691599
Philadelphia has a population of 1587761
Phoenix has a population of 1601381
San Antonio has a population of 1442472
San Diego has a population of 1397856
San Jose has a population of 1042940


Set the workspace to the geodatabase where the feature class was located

Set fc to the largest_cities feature class and set fields to the two fields being used

Created a cursor that searched through the specified feature class for the two fields I had previously set the fields variable to

If the row[1] variable, which corresponded to Population, was greater than 1 million then both the name and the population fields are printed out

## Problem 4

In [64]:
fc = "largest_cities"
fields = ["NAME", "ST", "CAPITAL"]

In [65]:
for row in sorted(arcpy.da.SearchCursor(fc, fields)):
    if row[2] == "State":
        print(row[0] + " is the capital of " + row[1])

Albany is the capital of NY
Annapolis is the capital of MD
Atlanta is the capital of GA
Augusta is the capital of ME
Austin is the capital of TX
Baton Rouge is the capital of LA
Bismarck is the capital of ND
Boise City is the capital of ID
Boston is the capital of MA
Carson City is the capital of NV
Charleston is the capital of WV
Cheyenne is the capital of WY
Columbia is the capital of SC
Columbus is the capital of OH
Concord is the capital of NH
Denver is the capital of CO
Des Moines is the capital of IA
Dover is the capital of DE
Frankfort is the capital of KY
Harrisburg is the capital of PA
Hartford is the capital of CT
Helena is the capital of MT
Honolulu is the capital of HI
Indianapolis is the capital of IN
Jackson is the capital of MS
Jefferson City is the capital of MO
Juneau is the capital of AK
Lansing is the capital of MI
Lincoln is the capital of NE
Little Rock is the capital of AR
Madison is the capital of WI
Montgomery is the capital of AL
Montpelier is the capital of VT

Set fc to the largest_cities feature class

Set the fields variable the the three values that tell the name, state the city is in, and whether or not its the capital

Cursor searches through the feature class
If row[2], which is the Capital field, says "State" that means its the state capital and it gets printed

## Problem 5

In [77]:
import os

In [78]:
arcpy.env.workspace = r"C:\Users\brend_000\Documents\ArcGIS\Projects\GIS 5090\GIS 5090 - Project 1\usa_cities.gdb"

In [79]:
fc = "largest_cities"
Copycities = arcpy.SelectLayerByAttribute_management(fc, "NEW SELECTION", '"POPULATION" > 1000000')

In [80]:
arcpy.CopyFeatures_management(Copycities, 'Copy_lyr')

<Result 'C:\\Users\\brend_000\\Documents\\ArcGIS\\Projects\\GIS 5090\\GIS 5090 - Project 1\\usa_cities.gdb\\Copy_lyr'>

Set the workspace to where the geodatabase containing the feature class file we want is

Set the fc variable to the feature class largest cities and set the Copycities variable to contain all the layers with attributes that have a population over 1 million

Copy the feature meeting that criteria by including Copycities in the parameters, so that only cities with over a million in population get carried over

## Bonus

In [3]:
arcpy.env.workspace = r"C:\Users\brend_000\Documents\ArcGIS\Projects\GIS 5090\GIS 5090 - Project 1\usa_cities.gdb"
fc = "largest_cities"
fields = ["NAME", "POPULATION"]

In [ ]:
for row in sorted(arcpy.da.SearchCursor(fc, fields)):